pandas/api/internals.py

```diff
@@ -0,0 +1,50 @@
+from pandas import DataFrame
```

Comment:
Can we add a docstring (module and/or function level) to the effect of "we discourage this for everyone except pyarrow. if you think you have a use case for this, let us know at [...]"

Reply:
@phofl might also have a use case in dask

Reply:
Yeah we are working on changing how we shuffle data were this would be helpful (we will get a huge number of small data frames, so overhead is painful), but I agree that we should strengthen this a little bit that makes it clear that end users shouldn't need this

Reply:
I already added a more generic note "For almost all use cases, you should use the standard pd.DataFrame(..) constructor instead." without naming specific libraries that use this. 

What would we gain with a "if you think you have a use case for this, let us know at"? Learning about use cases where people would use this is certainly valuable, but in the end it will be public developer API and so if we would in the future change or remove it, we need to go through normal deprecation processes anyway, I think.

Reply:
Hopefully we'll never have to revisit this again.  But if we do, there is evidence that discussions around a deprecation here would be more painful than elsewhere. It would be helpful to know ahead of such a discussion if anyone else was using it.  Moreover, the "let us know" is a chance to try to talk anyone out of using this.

Reply:
Added *"If you are planning to use this function, let us know by opening an issue at https://github.com/pandas-dev/pandas/issues."*